core/templates/pages/about-page/about-page.component.spec.ts

```diff
@@ -55,165 +59,186 @@ class MockI18nLanguageCodeService {
         },
+        {
+          provide: WindowDimensionsService,
+          useValue: {
+            isWindowNarrow: () => true
+          }
+        },
         { provide: TranslateService, useClass: MockTranslateService },
-        UtilsService,
+        { provide: SiteAnalyticsService, useValue: siteAnalyticsService },
         UrlInterpolationService,
-        { provide: WindowRef, useValue: windowRef }
-      ],
-      schemas: [NO_ERRORS_SCHEMA]
+        {
+          provide: WindowRef,
+          useValue: {
+            nativeWindow: {
+              location: {
+                href: ''
+              }
+            }
+          }
+        }
+      ]
     }).compileComponents();
-    fixture = TestBed.createComponent(AboutPageComponent);
-    component = fixture.componentInstance;
-  });
-
-  it('should click on about tab', () => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('about');
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#about');
-    expect(component.activeTabName).toBe('about');
-  });
-
-  it('should click on license tab', fakeAsync(() => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('license');
-    fixture.detectChanges();
-    expect(windowRef.nativeWindow.location.hash).toBe('#license');
-    expect(component.activeTabName).toBe('foundation');
-  }));
-
-  it('should click on foundation tab', () => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('foundation');
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#foundation');
-    expect(component.activeTabName).toBe('foundation');
   });
+  beforeEach(angular.mock.module('oppia'));
 
-  it('should click on credits tab', () => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('credits');
-
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#credits');
-    expect(component.activeTabName).toBe('credits');
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+    loaderService = TestBed.get(LoaderService);
+    userService = TestBed.get(UserService);
+    const aboutPageComponent = TestBed.createComponent(AboutPageComponent);
+    component = aboutPageComponent.componentInstance;
   });
-
-  it('should activate about tab on init', () => {
-    windowRef.nativeWindow.location.hash = '#about';
-
+  let component = null;
+
+  it('should successfully instantiate the component',
+    () => {
+      expect(component).toBeDefined();
+      component.ngOnInit();
+    });
+
+  it('should return correct static image url when calling getStaticImageUrl',
+    () => {
+      expect(component.getStaticImageUrl('/path/to/image')).toBe(
+        '/assets/images/path/to/image');
+    });
+
+  // eslint-disable-next-line max-len
```

Comment:
Don't use pragma.

`'should redirect guest user to the login page when they click' +
 'create lesson button'`

Reply:
done